com.github.mtkennerly.ludusavi.yaml

```
@@ -0,0 +1,52 @@
+app-id: com.github.mtkennerly.ludusavi
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+build-options:
+  append-path: /usr/lib/sdk/rust-stable/bin
+  env:
+    CARGO_HOME: /run/build/ludusavi/cargo
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  ## Needed for custom paths and restoring saves
+  - --filesystem=host
+  ## GPU acceration
+  - --device=dri
+  ## Fix for GUI being to large on the steam deck
+  - --env=WINIT_X11_SCALE_FACTOR=1
+command: ludusavi
+modules:
+  - name: ludusavi
+    buildsystem: simple
+    sources:
+      - type: git
+        url: https://github.com/mtkennerly/ludusavi
+        tag: v0.10.0
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/mtkennerly/ludusavi/latest
+          version-query: .tag_name
+          tag-query: .assets[] | select(.name==$version)
+            |  .tag_name
+      - generated-sources.json
+      - type: file
+        path: ludusavi.desktop
+      - type: file
+        path: com.github.mtkennerly.ludusavi.metainfo.xml
+    build-options:
+      env:
+        APPNAME: ludusavi
+    build-commands:
+      - cargo --offline fetch --verbose
+      - cargo build --offline --release
+      - install -D target/release/ludusavi /app/bin/ludusavi
+    post-install:
+      - install -Dm644 assets/icon.png /app/share/icons/hicolor/64x64/${FLATPAK_ID}.png
+      - install -Dm644 assets/icon.svg /app/share/icons/hicolor/scalable/${FLATPAK_ID}.svg
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - install -Dm644 ${APPNAME}.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - desktop-file-edit --set-key=Icon --set-value=${FLATPAK_ID} /app/share/applications/${FLATPAK_ID}.desktop
```

```suggestion
      - desktop-file-edit --set-icon${FLATPAK_ID} /app/share/applications/${FLATPAK_ID}.desktop
```

But as I mentionned, since you have the .desktop here, it might as well do it right directly.